conda/install.py

```diff
@@ -618,10 +619,20 @@ def link(pkgs_dir, prefix, dist, linktype=LINK_HARD, index=None):
         sys.exit('Error: pre-link failed: %s' % dist)
 
     info_dir = join(source_dir, 'info')
+
+    if not os.path.isfile(join(info_dir, "files")):
+        error_and_exit("Installing %s requires a minimum conda version of 4.3." % dist)
+
     files = list(yield_lines(join(info_dir, 'files')))
     has_prefix_files = read_has_prefix(join(info_dir, 'has_prefix'))
     no_link = read_no_link(info_dir)
 
+    with open(join(info_dir, "index.json")) as index_json:
+        data = json.load(index_json)
```

Comment:
Wait, no, this is wrong.  You don't need to do this here.  And in fact it probably won't be there.

Use `index.get('dist', {}).get('noarch', None) in (True, False, "generic")`